doc/user_guide/marks/geoshape.rst

```diff
@@ -0,0 +1,689 @@
+
+.. altair-plot::
+
+    from altair.expr import datum, geoCentroid
+
+    bubbles = alt.Chart(gdf_sel).transform_calculate(
+        centroid=geoCentroid(None, datum)
+    ).mark_circle(
+        stroke='black'
+    ).encode(
+        longitude='centroid[0]:Q',
+        latitude='centroid[1]:Q',
+        size="pop_est:Q" 
+    )
+
+    (basemap + bubbles).project(
+        type='identity', reflectY=True
+    )
+
+Lookup datasets
+~~~~~~~~~~~~~~~
+Sometimes your data is separated in two datasets. One ``DataFrame`` with the data and one ``GeoDataFrame`` with the geometries.
+In this case you can use the ``lookup`` transform to collect related information from the other dataset.
+
+You can use the ``lookup`` transform in two directions:
+
+1. Use a ``GeoDataFrame`` with geometries as source and lookup related information in another ``DataFrame``.
+2. Use a ``DataFrame`` as source and lookup related geometries in a ``GeoDataFrame``.
+
+Depending on your use-case one or the other is more favorable.
+
+First we show an example of the first approach. 
+Here we lookup the field ``rate`` from the ``df_us_unemp`` DataFrame, where the ``gdf_us_counties`` GeoDataFrame is used as source: 
+
+.. altair-plot::
+
+    import altair as alt
+    from vega_datasets import data
+    import geopandas as gpd
+
+    gdf_us_counties = gpd.read_file(data.us_10m.url, driver='TopoJSON', layer='counties')
+    df_us_unemp = data.unemployment()
+
+    alt.Chart(gdf_us_counties).mark_geoshape().transform_lookup(
+        lookup='id', 
+        from_=alt.LookupData(data=df_us_unemp, key='id', fields=['rate'])
+    ).encode(
+        alt.Color('rate:Q')
+    ).project(
+        type='albersUsa'
+    )
+
+Next, we show an example of the second approach.
+Here we lookup the geometries through the fields ``geometry`` and ``type`` from the ``gdf_us_counties`` GeoDataFrame, where the ``df_us_unemp`` DataFrame is used as source.
+
+.. altair-plot::
+
+    alt.Chart(df_us_unemp).mark_geoshape().transform_lookup(
+        lookup='id',
+        from_=alt.LookupData(data=gdf_us_counties, key='id', fields=['geometry', 'type'])
+    ).encode(
+        alt.Color('rate:Q')
+    ).project(
+        type='albersUsa'
+    )
+
+
+Chloropleth Classification
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+Chloropleth maps provide an easy way to visualize how a variable varies across a 
+geographic area or show the level of variability within a region. 
+
+We first define a utility function ``classify()`` that we will use to showcase different approaches to make a chloropleth map.
+We apply it to define a chloropleth map of the unemployment statistics of 2018 of US counties using a ``linear`` scale. 
+
+.. altair-plot::
+
+    import altair as alt
+    from vega_datasets import data
+    import geopandas as gpd
+
+    def classify(scale_type, breaks=None, nice=False, title=None, size='small'):
```

Comment:
This help function might seem a bit daunting for many readers, but I also get that it is convenient to define things upfront and reduce repetition later

Reply:
I had the same feeling when revisiting this PR. Will probably move this to the example gallery and simplify here.

Reply:
Simplified it using a single mark_geoshape without the jitter bar chart. Also removed the concatenated variant with all combinations. I feel its better suited in the example gallery.